app/uk/gov/hmrc/tai/viewModels/YourIncomeCalculationViewModel.scala

```diff
@@ -124,151 +127,175 @@ object YourIncomeCalculationViewModel {
-      getManualUpdateMsg(taxCodeIncome),
-      getSameMsg(employment, taxCodeIncome.amount, amountYearToDate, isPension, paymentDate),
-      getPayFreqMsg(employment, isPension, paymentFrequency, amountYearToDate, paymentDate, taxCodeIncome.amount)) match {
-
-      case ((Some(ceasedMsg), Some(ceasedMsgEstimate)), _, _, _) => (Some(ceasedMsg), Some(ceasedMsgEstimate))
-      case ((Some(ceasedMsg), None), _, _, _) => (Some(ceasedMsg), None)
-      case (_, (Some(manualUpdateMsg), Some(manualUpdateMsgEstimate)), _, _) => (Some(manualUpdateMsg), Some(manualUpdateMsgEstimate))
-      case (_, _, (Some(sameIncomeMsg), None), _) => (Some(sameIncomeMsg), None)
-      case (_, _, _, (Some(payFreqMsg), Some(payFreqMsgEstimate))) => (Some(payFreqMsg), Some(payFreqMsgEstimate))
-      case (_, _, _, (None, Some(payFreqMsgEstimate))) => (None, Some(payFreqMsgEstimate))
-      case _ => (Some(messages("tai.income.calculation.default." + pensionOrEmpMessage(isPension), Dates.formatDate(TaxYear().end))),
-        Some(messages("tai.income.calculation.default.estimate." + pensionOrEmpMessage(isPension), taxCodeIncome.amount)))
+                                employment: Employment,
+                                pensionOrEmployment: String,
+                                taxCodeIncome: TaxCodeIncome,
+                                paymentFrequency: Option[PaymentFrequency],
+                                amountYearToDate: BigDecimal,
+                                paymentDate: Option[LocalDate])(implicit messages: Messages): (Option[String], Option[String]) = {
+
+
+    lazy val ceasedIncomeMessages = (
+      ceasedIncomeCalculationMessage(employmentStatus, employment, pensionOrEmployment),
+      ceasedIncomeCalculationEstimateMessage(employmentStatus, employment, taxCodeIncome.amount)
+    )
+
+    lazy val manualIncomeMessages = {
+      val msg = manualUpdateIncomeCalculationMessage(taxCodeIncome)
+      (msg, if(msg.isDefined) manualUpdateIncomeCalculationEstimateMessage(taxCodeIncome) else None)
     }
-  }
 
-  def getCeasedMsg(employmentStatus: TaxCodeIncomeSourceStatus, employment: Employment, isPension: Boolean,
-                   amount: BigDecimal)(implicit messages: Messages): (Option[String], Option[String]) = {
+    lazy val sameMessages = (sameIncomeCalculationMessage(employment, taxCodeIncome.amount, amountYearToDate, pensionOrEmployment, paymentDate), None)
+
+    lazy val payFrequencyMessages = (
+      payFreqIncomeCalculationMessage(employment, pensionOrEmployment, paymentFrequency, amountYearToDate, paymentDate),
+      payFreqIncomeCalculationEstimateMessage(pensionOrEmployment, paymentFrequency, paymentDate, taxCodeIncome.amount)
+    )
 
-    (employmentStatus, employment.endDate.isDefined,
-      employment.cessationPay.isDefined) match {
+    Seq(ceasedIncomeMessages, manualIncomeMessages, sameMessages, payFrequencyMessages).find(fetchMessages.isDefinedAt) match {
```

Comment:
Instead of using the Partial function, we can just use this:
Seq(ceasedIncomeMessages, manualIncomeMessages, sameMessages, payFrequencyMessages).find(_ != (None,None)) match {